From 099b5d9ae313eca8e113cf14f4e7919e72817f28 Mon Sep 17 00:00:00 2001
From: andychou <andychou@google.com>
Date: Wed, 5 Dec 2018 19:18:09 +0800
Subject: [PATCH 18/39] Map ImsReasonInfo.CODE_SIP_USER_REJECTED(SIP 603
 DECLINE) as DisconnectCause.NORMAL

Network doesn't treat SIP 603 DECLINE as error condition
This can be observed from some carrier requirements like VZW
VZ_REQ_IMS_22961 or Softbank OTR-Voice-032026
So we should not treat it as server error.

Bug: b/119804523
Test: Build pass
Change-Id: I98d088120096a5686f788d85ecbc205e65475b15
---
 .../internal/telephony/imsphone/ImsPhoneCallTracker.java        | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/java/com/android/internal/telephony/imsphone/ImsPhoneCallTracker.java b/src/java/com/android/internal/telephony/imsphone/ImsPhoneCallTracker.java
index 7127eba62..a55d8745c 100644
--- a/src/java/com/android/internal/telephony/imsphone/ImsPhoneCallTracker.java
+++ b/src/java/com/android/internal/telephony/imsphone/ImsPhoneCallTracker.java
@@ -1993,6 +1993,7 @@ public class ImsPhoneCallTracker extends CallTracker implements ImsPullCall {
                 return DisconnectCause.INCOMING_REJECTED;
 
             case ImsReasonInfo.CODE_USER_TERMINATED_BY_REMOTE:
+            case ImsReasonInfo.CODE_SIP_USER_REJECTED:
                 return DisconnectCause.NORMAL;
 
             case ImsReasonInfo.CODE_SIP_FORBIDDEN:
@@ -2001,7 +2002,6 @@ public class ImsPhoneCallTracker extends CallTracker implements ImsPullCall {
             case ImsReasonInfo.CODE_SIP_REDIRECTED:
             case ImsReasonInfo.CODE_SIP_BAD_REQUEST:
             case ImsReasonInfo.CODE_SIP_NOT_ACCEPTABLE:
-            case ImsReasonInfo.CODE_SIP_USER_REJECTED:
             case ImsReasonInfo.CODE_SIP_GLOBAL_ERROR:
                 return DisconnectCause.SERVER_ERROR;
 
-- 
2.17.1

